bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/ui/view/visualizer/Visualizer.java

```diff
@@ -680,57 +681,57 @@ private void paintGroupBars(Graphics2D g2d) {
 	 * optionally as a closed shape corresponding to the area under the curve. These shapes
 	 * are cached.
 	 */
-	private Shape getElevationCurve(WorldCoords coords, long start, long end, boolean close, Map<WorldCoords, CachedShape> cache) {
+	private Shape getElevationCurve(Function<Long, WorldCoords> coords, long start, long end, boolean close, Map<WorldCoords, CachedShape> cache) {
 		synchronized (cacheLock) {
 
 			// Return the cached shape, if any.
-			CachedShape cs = cache.get(coords);
-			if (cs != null && cs.matches(start, end))
-				return cs.shape;
+//			CachedShape cs = cache.get(coords);
+//			if (cs != null && cs.matches(start, end))
+//				return cs.shape;
```

Comment:
If the plot without caching is quick enough we could get away without caching. In my machine it feels fine but haven't used the QPT in anger.

Since functions can't be cached, we'll have to pick the coordinates at certain times as you mention. But can we draw the same exact non-cached plot with just those few coordinates? Is it alright to have an approximation for the elevation plot when cached? Maybe we could cache every step used to draw the plot but that may make the cache too large.

In any case I think we should reassess first if the performance is good enough without caching.